.github/workflows/BuildDockerContainer.yaml

```diff
@@ -0,0 +1,61 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# This workflow can be run manually on any branch to build and push our Docker
+# containers.
+name: Build Docker container
+
+on:
+  workflow_dispatch:
+    inputs:
+      image_name:
+        description: >
+          Image name to push to DockerHub
+        required: true
+        default: 'sxscollaboration/spectre'
+
+jobs:
+  docker:
+    name: Build and deploy
+    runs-on: ubuntu-latest
+    environment: containers
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      # Needed once we do multi-platform builds
+      # - name: Set up emulation support
+      #   uses: docker/setup-qemu-action@v2
+      #   with:
+      #     platforms: arm64
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      # We could also push containers to GitHub instead of DockerHub
+      # - name: Login to GitHub container registry
+      #   uses: docker/login-action@v2
+      #   with:
+      #     registry: ghcr.io
+      #     username: ${{ github.actor }}
+      #     password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build dev container
+        uses: docker/build-push-action@v4
+        with:
+          push: true
```

Comment:
Is there a way to have this wait for another approval before pushing? Sort of like our releases? Just so we can inspect things if need be

Reply:
Waiting for another approval between build and push would be more complicated because it needs to be split into separate jobs. See the `environment` option in line 21. We can configure protection rules for the environment, e.g. that it requires an approval, but it applies to the whole job.

I intend to use this on a branch to push to my own Dockerhub (or some other staging area) by setting the `image_name` to something like `nilsvu/spectre`. Then I can test it before pushing to `sxscollaboration/spectre` manually. I'll also try to automate the testing step before pushing, but in a separate PR.

Reply:
Do you intend to use this workflow on develop then (in addition to your fork)?

Reply:
Probably not? Usually we want to push the container to a staging area before merging a PR.

Reply:
If we don't plan on using this on develop really, is there a way to only add this to your fork?

Reply:
No I think adding it to the repo is the right thing to do. It allows anyone of us to trigger the build from the PR branch. And it allows to run the tests over the new container (see new commit).